exonum/src/sandbox/consensus/block_request.rs

```diff
@@ -304,6 +304,85 @@ fn handle_block_response_with_incorrect_tx() {
     // an incorrect tx, node should panic.
 }
 
+/// - should **NOT** process block if tx is incorrect
+/// idea of test is:
+/// - getting Status from other node with later height, send BlockRequest to this node
+/// - receive incorrect tx
+/// - receive BlockResponse with known tx A
+/// - Figure out that block is incorrect
+/// - Node should panic because of committed block with incorrect tx.
+#[test]
+#[should_panic(expected = "Received a block with transaction known as invalid")]
+fn handle_block_response_with_known_incorrect_tx() {
+    let sandbox = timestamping_sandbox();
+
+    // Create correct tx, and then sign with the wrong destination.
+    let incorrect_tx = gen_incorrect_tx();
+    let propose = ProposeBuilder::new(&sandbox).build();
+    let block = sandbox.create_block(&[incorrect_tx.clone()]);
+
+    let precommit_1 = sandbox.create_precommit(
```

Comment:
Nit: Since precommits are used in bulk, it could make sense to create them like this:

```rust
let precommits: Vec<_> = (1..=3)
    .map(|id| {
        sandbox.create_precommit(/* ... */)
    })
    .collect();
```